store/driver/txn/error.go

```diff
@@ -100,6 +125,8 @@ func extractKeyExistsErrFromHandle(key kv.Key, value []byte, tblInfo *model.Tabl
 		if col.Length > 0 && len(str) > col.Length {
 			str = str[:col.Length]
 		}
+
+		str = fmtNonASCIIPrintableCharToHex(str)
```

Comment:
We need to convert it only if the column type is binary. See `types.IsBinaryStr()`.

```sql
mysql> insert into t1 values ('测试');
Query OK, 1 row affected (0.06 sec)

mysql> insert into t1 values ('测试');
ERROR 1062 (23000): Duplicate entry '测试' for key 't1.PRIMARY'
mysql> select version();
+-----------+
| version() |
+-----------+
| 8.0.27    |
+-----------+
1 row in set (0.02 sec)
```